src/actions/router.js

```diff
@@ -1,14 +1,55 @@
 import parseUrl from 'grape-web/lib/parse-url'
 import * as router from 'react-router-redux'
 import find from 'lodash/collection/find'
+import {matchPath} from 'react-router-dom'
 
 import conf from '../conf'
 import * as types from '../constants/actionTypes'
 import {channelsSelector} from '../selectors'
 import {findLastUsedChannel} from './utils'
+import {channelRoute, pmRoute} from '../constants/routes'
+import {isChatUrl} from '../components/grapedown/utils'
 
 import {setChannel, openPm, openChannel, handleBadChannel} from './'
 
+const goToFromEmbedded = path => (dispatch) => {
+  const {pathname, hostname} = parseUrl(path)
+  const match = matchPath(pathname, {path: channelRoute}) ||
+    matchPath(pathname, {path: pmRoute})
+  // Open link in new tab, if it's outgoing link, or if link not leads to chat.
```

Comment:
We need  to open a new tab if the path path doesn't match those routes in embedded mode.